main.cpp

```diff
@@ -119,6 +140,43 @@ void render_scene(color *display, size_t width, size_t height,
     }
 }
 
+const Scene load_scene_from_file(const char* filename)
+{
+   std::ifstream infile(filename);
+   Scene scene = {};
+   std::string type, line;
+
+   while(std::getline(infile, line)) {
+	  std::istringstream iss(line);
+	  iss >> type;
+
+	  if(type.compare("e") == 0) // eye
```

Comment:
`std::string` overloads `operator==`.

Please replace all of the `type.compare("string")` with `type == "string"`